File: src/menu/Menu.java
package menu;

import productos.Alimento;
import productos.Belleza;
import productos.Electrodomestico;
import productos.Limpieza;
import tienda.Tienda;
import usuarios.Cliente;

import java.util.Scanner;

public class Menu {
    public static Scanner sc = new Scanner(System.in);
    public static Tienda tienda = new Tienda();

    public  static void MenuPrincipal(){
        boolean banSalir = true;
        System.out.println("*** BIENVENIDO A MI CHILCHOTA ***");
        do {
            System.out.println();
            System.out.println("SELECCIONA UNA OPCION: ");
            System.out.println("1. Registrar un producto nuevo. ");
            System.out.println("2. Registrar a un cliente nuevo. ");
            System.out.println("3. Comprar un producto.");
            System.out.println("4. Ver todos los productos comprados. ");
            System.out.println("5. Ver todas las compras de un cliente en especifico. ");
            System.out.println("6. Eliminar un producto.");
            System.out.println("7. Eliminar a un cliente. ");
            System.out.println("8. Ver el numero de clientes. ");
            System.out.println("9. Ver el numero de productos. ");
            System.out.println("10. Mostrar a todos los productos: ");
            System.out.println("11. SALIR");

            int opcion = sc.nextInt();

            switch (opcion) {
                case 1:
                    registrarProducto();
                    break;
                case 2:
                    registrarCliente();
                    break;
                case 3:
                   comprarProducto();
                    break;
                case 4:
                    Tienda.productosComprados();
                    break;
                case 5:
                   comprasUsuarioEspecifico();
                    break;
                case 6:
                    eliminarProducto();
                    break;
                case 7:
                    eliminarCliente();
                    break;
                case 8:
                    Tienda.numeroClientes();
                    break;
                case 9:
                    Tienda.numeroProductos();
                    break;
                case 10:
                    Tienda.mostrarTodosLosProductosPorCategoria();
                    break;
                case 11:
                    System.out.println("Saliendo del sistema.");
                    banSalir = false;
                    break;
                default:
                    System.out.println("OPCION INVALIDA");
                    break;
            }

        } while (banSalir);
    }
    public static void inisiarSesion(){

        String password;
        int cont = 1;
        System.out.println("Bienvenido al sistema del mi chilchota, por favor ingresa la contraseña para poder acceder al menú, tienes 5 intentos");
        do {
            System.out.println("intento numero: " + cont);
            password = sc.nextLine();
            cont++;
            if (cont == 6) {
                System.out.println("A sobre pasado el  numero de intentos");
                return;
            }
        } while (!Tienda.checkPaswword(password));

        MenuPrincipal();
    }

    public static void registrarProducto() {
        sc.nextLine();
        int opc;
        System.out.println("Ingresa el nombre del producto: ");
        String nombre = sc.nextLine();
        System.out.println("Ingresa el precio del producto: ");
        float precio = sc.nextFloat();
        sc.nextLine();
        System.out.println("Ingresa la fecha de importacion  XX/XX/XXXX: ");
        String fechaImport = sc.nextLine();
        System.out.println("Ingresa el numero de serie: ");
        int numSerie = sc.nextInt();
        System.out.println("Ingresa el stock deseado del producto: ");
        int stock = sc.nextInt();
        sc.nextLine();
        do {
            System.out.println("Selecciona el tipo de producto a registrar: ");
            System.out.println("1. Electrodomesticos");
            System.out.println("2. Limpieza");
            System.out.println("3. Belleza");
            System.out.println("4. Alimentos");
            opc = sc.nextInt();
            sc.nextLine();
        } while (opc < 1 || opc > 4);

        switch(opc) {
            case 1:
                System.out.println("Ingresa la marca del electrodomestico: ");
                String marca = sc.nextLine();
                tienda.addElectrodomestico(new Electrodomestico(nombre,precio, fechaImport, numSerie, stock, marca));

                break;
            case 2:
                System.out.println("Ingresa el aroma del producto: ");
                String aroma = sc.nextLine();
                tienda.addLimpieza(new Limpieza(nombre, precio, fechaImport, numSerie, stock,aroma));

                break;
            case 3:
                System.out.println("Ingresa el tono de piel: ");
                String tonoPiel = sc.nextLine();
                tienda.addBelleza(new Belleza(nombre, precio, fechaImport, numSerie, stock, tonoPiel));
                break;
            case 4:
                System.out.println("Ingresa la fecha de caducidad: ");
                String fechaCaducidad = sc.nextLine();
                tienda.addAlimeto(new Alimento(nombre, precio, fechaImport, numSerie, stock, fechaCaducidad));
                break;
        }

    }
    public static void registrarCliente(){
        sc.nextLine();

        System.out.println("Ingresa el nombre del cliente: ");
        String nombre = sc.nextLine();
        System.out.println("Ingresa el apellido del cliente:");
        String apellido = sc.nextLine();
        System.out.println("Ingresa la edad del cliente: ");
        int edad = sc.nextInt();
        sc.nextLine();
        tienda.addCliente(new Cliente(nombre,apellido,edad));
    }

    public static void comprarProducto(){
        Tienda.mostrarCliente();
        System.out.println("Selecciona el cliente con el que deseas comprar un producto: ");
        int indexCliente=sc.nextInt();
        System.out.println("*** Productos Disponibles ***");
        Tienda.mostrarTodosLosProductos();
        System.out.println("Ingresa el indice del producto que deseas comprar: ");
        int indexCompra = sc.nextInt();
        Tienda.comprarProducto(indexCliente,indexCompra);
    }
    public static void comprasUsuarioEspecifico(){
        Tienda.mostrarCliente();
        System.out.println("Digite el indice del Cliente: ");
        int indexCliente= sc.nextInt();
        Cliente cliente=Tienda.clientes.get(indexCliente);
        System.out.println("Compras de "+cliente.getName());
        cliente.mostrarCompras();
    }

    public static void eliminarProducto(){
        Tienda.mostrarTodosLosProductos();
        System.out.println("Digite el indice del producto a remover");
        int indexRemover=sc.nextInt();
        Tienda.removerProducto(indexRemover);
    }
    public static void eliminarCliente(){
        Tienda.mostrarCliente();
        System.out.println("Digite el indice del cliente a remover");
        int indexRemover=sc.nextInt();
        Tienda.removerCliente(indexRemover);
    }
}

File: src/menu/Menu.py
from productos.Alimento import Alimento
from productos.Belleza import Belleza
from productos.Electrodomestico import Electrodomestico
from productos.Limpieza import Limpieza
from tienda.Tienda import Tienda
from usuarios.Cliente import Cliente

tienda = Tienda()


class Menu:

    @staticmethod
    def iniciar_sesion():
        cont = 1
        print("Bienvenido al sistema del mi chilchota,"
              " por favor ingresa la contraseña para poder acceder al menú, "
              "tienes 5 intentos")
        while cont <= 5:
            print("Intento número:", cont)
            password = input()
            cont += 1
            if cont == 6:
                print("Has superado el número de intentos.")
                return
            if tienda.check_paswword(password):
                Menu.menu_principal()
                return
        return

    @staticmethod
    def menu_principal():
        print("*** BIENVENIDO A MI CHILCHOTA ***")
        ban_salir = True
        while ban_salir:
            print("\nSELECCIONA UNA OPCION: ")
            print("1. Registrar un producto nuevo.")
            print("2. Registrar a un cliente nuevo.")
            print("3. Comprar un producto.")
            print("4. Ver todos los productos comprados.")
            print("5. Ver todas las compras de un cliente en específico.")
            print("6. Eliminar un producto.")
            print("7. Eliminar a un cliente.")
            print("8. Ver el número de clientes.")
            print("9. Ver el número de productos.")
            print("10. Mostrar todos los productos.")
            print("11. SALIR")

            opcion = int(input())

            if opcion == 1:
                Menu.registrar_producto()
            elif opcion == 2:
                Menu.registrar_cliente()
            elif opcion == 3:
                Menu.comprar_producto()
            elif opcion == 4:
                tienda.productos_comprados()
            elif opcion == 5:
                Menu.compras_usuario_especifico()
            elif opcion == 6:
                Menu.eliminar_producto()
            elif opcion == 7:
                Menu.eliminar_cliente()
            elif opcion == 8:
                tienda.numero_clientes()
            elif opcion == 9:
                tienda.numero_productos()
            elif opcion == 10:
                tienda.mostrar_todos_los_productos_por_categoria()
            elif opcion == 11:
                print("Saliendo del sistema.")
                ban_salir = False
            else:
                print("OPCIÓN INVÁLIDA")

    @staticmethod
    def registrar_producto():
        opc = 0
        while opc < 1 or opc > 4:
            print("Selecciona el tipo de producto a registrar: ")
            print("1. Electrodomesticos")
            print("2. Limpieza")
            print("3. Belleza")
            print("4. Alimentos")
            opc = int(input())

        nombre = input("Ingresa el nombre del producto: ")
        precio = float(input("Ingresa el precio del producto: "))
        fecha_import = input("Ingresa la fecha de importación XX/XX/XXXX: ")
        num_serie = int(input("Ingresa el número de serie: "))
        stock = int(input("Ingresa el stock deseado del producto: "))

        if opc == 1:
            print("Ingresa la marca del electrodomestico: ")
            marca = input()
            tienda.add_electrodomestico(Electrodomestico(nombre, precio, fecha_import, num_serie, stock, marca))

        elif opc == 2:
            print("Ingresa el aroma del producto: ")
            aroma = input()
            tienda.add_limpieza(Limpieza(nombre, precio, fecha_import, num_serie, stock, aroma))

        elif opc == 3:
            print("Ingresa el tono de piel: ")
            tono_piel = input()
            tienda.add_belleza(Belleza(nombre, precio, fecha_import, num_serie, stock, tono_piel))

        elif opc == 4:
            print("Ingresa la fecha de caducidad: ")
            fecha_caducidad = input()
            tienda.add_alimento(Alimento(nombre, precio, fecha_import, num_serie, stock, fecha_caducidad))

    @staticmethod
    def registrar_cliente():
        nombre = input("Ingrese el nombre del cliente: ")
        apellido = input("Ingresa el apellido del cliente:")
        edad = input("Ingresa el edad del cliente")
        tienda.add_cliente(Cliente(nombre, apellido, edad))

    @staticmethod
    def comprar_producto():
        tienda.mostrar_clientes()
        index_cliente = input("elecciona el cliente con el que deseas comprar un producto: ")
        print("*** Productos Disponibles ***")
        tienda.mostrar_todos_los_productos()
        index_producto = input("Ingresa el indice del producto que deseas comprar:")
        tienda.comprar_productos(index_cliente, index_producto)

    @staticmethod
    def compras_usuario_especifico():
        tienda.mostrar_clientes()
        index_cliente = input("Digite el indice del Cliente: ")
        cliente = tienda.get_cliente(index_cliente)
        print("Compras de ", cliente.get_name())
        cliente.mostrar_compras()

    @staticmethod
    def eliminar_producto():
        tienda.mostrar_todos_los_productos()
        index_remover = input("Digite el indice del producto a remover")
        tienda.remover_producto(index_remover)

    @staticmethod
    def eliminar_cliente():
        tienda.mostrar_clientes()
        index_remover = input("Digite el indice del cliente a remover")
        tienda.remove_cliente(index_remover)
